addons/im_livechat/static/src/core/web/suggestion_service_patch.js

```diff
@@ -11,8 +11,15 @@ patch(SuggestionService.prototype, {
             ? [...super.getSupportedDelimiters(...arguments), [":"]]
             : super.getSupportedDelimiters(...arguments);
     },
+    async fetchSuggestions({ delimiter, term }, { thread } = {}) {
+        if (thread.type === "livechat" && delimiter === "#") {
+            return;
+        }
+        return super.fetchSuggestions(...arguments);
+    },
```

Comment:
Ok for a fix. Could help to store logic in a delimiter object (more complex than just the char and optional start index), so that there's a function on delimiter to tell whether the delimiter is active or not on a given thread.

Reply:
It is indeed a good suggestion. I would like to do this as a seperate task in master as `ref` or `imp`. I've created a task in odoo/web to follow up.